src/js/system/RoutingService.js

```diff
@@ -0,0 +1,113 @@
+import {Route} from 'react-router';
+
+const queue = {};
+
+class Tab {
+  constructor(path, component) {
+    this.path = path;
+    this.component = component;
+  }
+
+  getPath() {
+    return this.path;
+  }
+
+  getComponent() {
+    return this.component;
+  }
+}
+
+class Page {
```

Comment:
As you're using classes to define the different rout types I would suggest to introduce a generic rout type.